src/plugins/shs.ts

```diff
@@ -22,6 +22,11 @@ export = {
   name: 'multiserver-shs',
   version: '1.0.0',
   init (api: any, config: Config) {
+    const keys = config.keys && toSodiumKeys(config.keys)    
+    if (!keys) {
+      console.error(new Error('Config object should contains SHS keys'))
+    }
```

Comment:
I like this. I would hazard you should probably throw instead of printing an error.

I'd also like to see a test around this - you may find you never see this log because toSodium *could* error loudly if things go wrong internally to that? (haven't looked at it)

Reply:
@mixmix appreciate the review. the change has been made to throw rather than log. trying to think through the testing changes needed and could use some additional feedback.

this is what I've come up with so far to add a test case using the `shs` plugin directly, as catching the error when creating a new server catches the missing keys error but throws again when `core.js` goes to load and the `shs` transform is missing. Would the following suffice?
```js
tape('throw error when no keys are supplied', function (t) {
  var shsPlugin = require('../lib/plugins/shs')
  var api = {}
  var config = {}

  // can't use create() here as catching the error for missing
  // keys causes another error downstream in ../lib/core where
  // the shs plugin is missing 'secret-stack needs at least 1 
  // transform protocol'
  try {
    shsPlugin.init(api, config)
  } catch (err) {
    t.ok(err, err.message)
  } 
  t.end()
})
```

Should this PR be enhanced to throw only when both keys and a seed are missing? That would save us from updating all the tests.

cc @staltz

Reply:
I am not familiar enough with this code but expect that both keys need to be there... Or at least the public key.

Your test looks really scoped and sufficient to me. BTW you have essentially recreated tapes t.throws


You could write it like this

```
t.throws(
  () => plugin.init(ssb, config), 
  /shs plugin requires valid config.keys/
  "throws without keys"
) 
```

Seconds arg is a regex which will be used to test `error.message` on the error which is thrown

Reply:
This has been updated to include 2 new tests. One which throws an error when neither keys nor a seed are passed into with the config. A second test is included that shows the publicKey is being correctly populated when passing just a seed. 

This is usually accomplished in [multiserver](https://github.com/ssb-js/multiserver/blob/7d47892757c14732b47662d81e36dd21f3c16b6b/plugins/shs.js#L9) where it would use the keys and expose the public key.   Unfortunately we can't wait for this to make it down to multiserver when throwing for missing keys at this level, since the publicKey is needed right away. 

This seems like a good idea to preserve some backward compatibility regarding the use of seeds, but open to being told otherwise :). This also introduces another dependency, `secret-handshake`, to use the `toKeys` method to [convert the seed to a keypair](https://github.com/austinfrey/secret-stack/blob/app-key-should-not-be-undefined/src/plugins/shs.ts#L27) but maybe using `chloride` directly would make more sense.